src/model.js

```diff
@@ -51,35 +51,32 @@ const Model = function(_data) {
         return resultProxy;
     }
     // Watch changes via proxy
-    this.data = newProxy(_data, '');    
+    this.data = newProxy(_data, '');
 }
+
+const jsPathRegex = /([^[.\]])+/g;
+
 // get attribute
-Model.prototype.get = function(key, fallback = null)
+Model.prototype.get = function(key, fallback = undefined)
 {
-    let data = this.data, keys = key.split('.');
-    // Traverse dots
-    for (let k =0; k < keys.length; k++) {
-        // Fallback if key missing
-        if (!(keys[k] in data)) return fallback;
-        data = data[keys[k]];
-    }
-    return data;
+    if (!key) return fallback;
+
+    const keyArray = Array.isArray(key) ? key : key.match(jsPathRegex);
+
+    return (
+        keyArray.reduce((prevObj, key) => prevObj && prevObj[key], this.data) || fallback
+    );
 }
 // Set attribute
 Model.prototype.set = function(key, value)
 {
-    let data = this.data, keys = key.split('.');
-    // Traverse dots
-    for (let k =0; k < keys.length-1; k++) {
-        // Fallback if key missing
-        if (!(keys[k] in data)){
-            data = data[keys[k]] = {};
-        } else {
-            data = data[keys[k]];
-        }
-    }
-    // Set last key manually as primatives will copy
-    return data[keys[keys.length-1]] = value;
+    const keyArray = Array.isArray(key) ? key : key.match(jsPathRegex);
+
+    keyArray.reduce((acc, prop, i) => {
+        if (acc[prop] === undefined) acc[prop] = {}

```

Comment:
; ?

Reply:
Ergh, yes. Need a linter or prettier config running here haha.

```suggestion
        if (acc[prop] === undefined) acc[prop] = {};
```